specs/Spec_for_adding_user.adoc

```diff
@@ -0,0 +1,166 @@
+# User Management: Adding new User in Tendrl
+
+## Problem description
+
+This spec talks about Adding/Creating a new User to Tendrl by the Admin
+
+## Use Cases
+
+The flow goes as follows:-
+
+* A user login as Admin
+* Clicks on Admin tab in Navigation
+* The navigation drops down
+* Clicks on *Users*
+* Clicks on *Add* button in Users page view
+
+User will land onto https://redhat.invisionapp.com/share/KNB25OEQT#/screens/226063802[Add Users] page
+
+## Proposed change
+
+### UI changes
+
+* Add *Users* tab in Admin navigation tab
+* Add *Add User* Button in Users Page view
+* Add a form for adding a user. The form will contain following feilds:-
+** User ID
+*** Input type : text
+*** Validations:
+**** Should not contain special characters
+**** Should be Unique
+**** Can be of a particular Maximum length
+** First Name
+*** Input type : text
+*** Validations:
+**** Should not contain special characters
+**** Can be of a particular Maximum length
+** Last Name
+*** Input type : text
+*** Validations:
+**** Should not contain special characters
+**** Can be of a particular Maximum length
+** Email
+*** Input type : email
+*** Validations:
+**** Validate email ID
+** Email Notification
+*** Input type : checkbox
+** Password
+*** Input type : password
+** Confirm Password
+*** Input type : password
+** Set a Role
+*** Admin
+*** Normal user
+*** Limited user
+*** Input type : Radio button
+
+
+### Impacted Modules:
+
+#### Tendrl API impact:
+
+```
+POST /api/1.0/users - creates a new user
+```
+
+* Sample API response now : 
+
+```
+{
+	"email": "thardy@tendrl.org",
+	"username": "thardy",
+	"name": "Tom Hardy",
+	"role": "admin"
+}
+
+```
+
+* Sample API response:
+
+```
+{
+	"username": "JDoe1",
+	"first_name": "John",
+	"last_name": "Doe",
+	"email": "jdoe@tendrl.org",
+	"email_notifications": "true",
+	"role": "admin"
+}
+
+```
+
+API support present for -
+
+* email
+* username
+* name
+* role
+
+Required from API - 
+
+* first_name
+* last_name
+* email_notifications
+
+#### Notifications/Monitoring impact:
+
+If ```email_notifications = true ```, email notifications are required to be monitored 
+
+
+### Security impact:
+
+### Other end user impact:
+
+None
+
+### Performance impact:
+
+None
+
+### Other deployer impact:
+
+None
+
+### Developer impact:
+
+None
+
+## Implementation:
+
+
+### Assignee(s):
+
+Primary assignee:
+  gnehapk
+
+Other contributors:
+  a2batic
+  
+### Work Items:
+
+https://github.com/Tendrl/specifications/issues/176
+
+
+## Dependencies:
+
+Need API support for required additional fields:
+
+* first_name
+* last_name
+* email_notifications
+
+
+## Testing:
+
+Test whether fields of adding user form are validating.
```

Comment:
Check if user is actually getting created and they are able to login to the system and UI is displayed as per their role :)